### PR TITLE
DO NOT MERGE: Revert "initial commit"

### PR DESCRIPTION
**Pull request by ethanpailes**
This reverts commit e23aee70e82f224eab6529506799ee9d94929673.

This is just to give us a diff to hang review comments off of, I'll fix them with followups.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -1,95 +0,0 @@
...
-fn run() -> Result<(), Error> {
-    println!("cargo:rerun-if-changed=src/env_shim.c");
-    // println!("cargo:rerun-if-changed=src/pam_motd_overlay_versions.ldscript");
-
-    let out_dir = env::var("OUT_DIR").map_err(|e| berr!("no OUT_DIR: {}", e))?;
-
-    let mut target_so = PathBuf::from(out_dir);
-    target_so.push("env_shim.so");
-
-    let cc = select_cc()?;
```
Comment (Aetf) [inline build.rs:38]: FYI https://docs.rs/cc/latest/cc/

AFAIK that's the defacto standard way of building c source files in build.rs

File: src/lib.rs [hunk 2]
```diff
@@ -1,165 +0,0 @@
...
-/// Errors encountered while setting up the environment injector.
-#[non_exhaustive]
-#[derive(Debug)]
-pub enum Error {
-    /// An opaque error with a useful debugging message but
-    /// which callers should not dispatch on.
-    Err {
-        msg: String,
-    },
-    __NonExhaustive,
```
Comment (Aetf) [inline src/lib.rs:35]: Why do we need this given the `non_exhaustive` attribute?

**Review by Aetf — APPROVED** [review]

File: build.rs (hunk 1 above)
Comment (ethanpailes) [reply build.rs:38]: Yeah that was the first thing I tried (not for the particular crate the the first time I did some cursed LD_PRELOAD shennaigans in rust as part of the motd crate), but unfortunately it does not support producing .so files. I think it only produces .a files.

File: src/lib.rs (hunk 2 above)
Comment (ethanpailes) [reply src/lib.rs:35]: Oh, maybe I can just leave it off.

Comment (ethanpailes) [reply src/lib.rs:35]: #5

**Comment by ethanpailes** [comment]
Fixes applied

**Pull request closed without merging**